src/signalrclient/json_helpers.cpp

```diff
@@ -45,6 +45,68 @@ namespace signalr
         }
     }
 
+    char getBase64Value(uint32_t i)
+    {
+        char index = (char)i;
+        if (index < 26)
+        {
+            return 'A' + index;
+        }
+        if (index < 52)
+        {
+            return 'a' + (index - 26);
+        }
+        if (index < 62)
+        {
+            return '0' + (index - 52);
+        }
+        if (index == 62)
+        {
+            return '+';
+        }
+        if (index == 63)
+        {
+            return '/';
+        }
+
+        throw std::out_of_range("base64 table index out of range: " + std::to_string(index));
+    }
+
+    std::string base64Encode(const std::vector<uint8_t>& data)
+    {
+        std::string base64result;
+
+        size_t i = 0;
+        while (i <= data.size() - 3)
+        {
+            uint32_t b = ((uint32_t)data[i] << 16) | ((uint32_t)data[i + 1] << 8) | (uint32_t)data[i + 2];
+            base64result.push_back(getBase64Value(b >> 18));
```

Comment:
```suggestion
            base64result.push_back(getBase64Value(b >> 18 & 0x3F));
```

Or remove it from the other places it's unnecessary. Or don't.